index.bs

```diff
@@ -316,56 +317,96 @@ Interface {#interface}
 =================
 
 <pre class="idl">
-dictionary NavigatorUABrandVersionDict {
-  required DOMString brand;
+dictionary NavigatorUABrandVersion {
+  DOMString brand;
   DOMString version;
 };
 
+dictionary UADataValues {
+  DOMString platform; 
+  DOMString platformVersion;
+  DOMString architecture;
+  DOMString model;
+};
+
 [Exposed=Window]
 interface NavigatorUAData {
-  readonly attribute FrozenArray&lt;NavigatorUABrandVersionDict&gt; brand;
+  readonly attribute FrozenArray&lt;NavigatorUABrandVersion&gt; uaList;
   readonly attribute boolean mobile;
-  Promise&lt;DOMString&gt; getPlatform();
-  Promise&lt;SOMString&gt; getPlatformVersion();
-  Promise&lt;DOMString&gt; getArchitecture();
-  Promise&lt;DOMString&gt; getModel();
+  Promise&lt;UADataValues&gt; getHighEntropyValues(sequence&lt;DOMString&gt; hints);
 };
 
 interface mixin NavigatorUA {
-  [SecureContext] NavigatorUAData getUserAgent();
+  [SecureContext] readonly attribute NavigatorUAData userAgentData;
 };
 Navigator includes NavigatorUA;
 
 </pre>
 
 Processing model {#processing}
 --------------
-<dfn method for="NavigatorUA"><code>getUserAgent()</code></dfn> method MUST run these steps:
+
+<h3 id="monkeypatch-html-windoworworkerglobalscope"><code>WindowOrWorkerGlobalScope</code></h3>
+
+Each user agent has an associated <dfn>UA list</dfn>, which is a [=/list=] created by running [=create UA list=].
+
+Every {{WindowOrWorkerGlobalScope}} object has an associated <dfn for="WindowOrWorkerGlobalScope">UA frozen array</dfn>, which is a <code><a interface>FrozenArray</a>&lt;<a dictionary>NavigatorUABrandVersion</a>></code>. It is initially the result of [=create a frozen array|creating a frozen array=] from [=UA list=].
+
+<h3 id="create-ua-list-section">Create a UA list</h3>
+
+When asked to run the <dfn>create UA list</dfn> algorithm, the user agent MUST run the following steps:
```

Comment:
Nit: the spec is inconsistent about whether it uses MUST or must. As the editor it's your call, but I personally prefer following https://infra.spec.whatwg.org/#conformance and sticking with lowercase.

Reply:
I'll fix that and the `[=user agent=]` comment everywhere in a followup